### PR TITLE
don't bother caching if we're using a nil repo

### PR DESCRIPTION
**Pull request by Stebalien**
(it'll be wrong anyways)

### CODE REVIEW

File: core/builder.go [hunk 1]
```diff
@@ -205,22 +205,24 @@ func setupNode(ctx context.Context, n *IpfsNode, cfg *BuildCfg) error {
...
-		return err
+	if !cfg.NilRepo {
+		bs, err = bstore.CachedBlockstore(ctx, bs, opts)
+		if err != nil {
+			return err
+		}
 	}
 
-	wbs = bstore.NewIdStore(wbs)
+	bs = bstore.NewIdStore(bs)
```
[the remark this answers is not in the document]
Comment (kevina) [reply core/builder.go]: That's fine.

**Comment by kevina** [comment]
@Stebalien what's the motivation for this change. When is a nil repo used?

**Comment by Stebalien** [comment]
> @Stebalien what's the motivation for this change. When is a nil repo used?

When adding files with `--only-hash`.

**Comment by Stebalien** [comment]
The main reasons are:

1. It doesn't do anything useful when we have a nil repo.
2. It may actually record bad information when using a nil repo (because the repo will throw everything away).

In practice, given where we use the nil repo, this really shouldn't make much of a difference (maybe a slight performance gain).

**Pull request merged**